Describe the spoofed source address attack. (#23)
marten-seemann/draft-seemann-quic-address-discovery: draft-seemann-quic-address-discovery.md

and observing if the responses are consistent). This logic is out of scope for
this document.

## On the Responder Side {#responder-side-security}

Depending on the routing setup, a node might not be able to observe the peer's
reflexive transport address, and attempts to do so might reveal details about
the internal network. In these cases, the node SHOULD NOT offer to provide
address observations.

> On path attackers could capture "normal" packets sent from requester to

Reviewer: ```suggestion
On-path attackers could capture packets sent from the requester to the
```

> responder, and resend them from a spoofed source address. Such spoofed
> packets could trigger production of spurious OBSERVED_ADDRESS frames.

Reviewer: ```suggestion
responder, and resend them from a spoofed source address. If done repeatedly,
these spoofed packets could trigger the sending of a large number of OBSERVED_ADDRESS frames.
```

The recommendation to only include OBSERVED_ADDRESS frames in packets
sent on the same path over which the address was observed offers a first

> layer of protection, as these packets will not reach the requester if

Reviewer: How does this protect against the sender?

Author: If sent over the same path, they will only reach the peer if the path is valid, thus will not have an effect on the peer. So, no pollution. But yes, still a volume effect. Maybe that could be said differently

the path is not valid. The attack also has the effect of causing spurious
detection NAT rebinding. We expect that implementations of QUIC have sufficient
protection against spurious NAT rebinding to limit the incidental traffic
caused by such attacks, and will not produce a high volume of

> spurious OBSERVED_ADDRESS frames.

Reviewer: Should we link to the section in RFC 9000 that describes this attack?

Author: Yes.

Author: Check the most recent commit please.

and observing if the responses are consistent). This logic is out of scope for
this document.

## On the Responder Side {#responder-side-security}

Depending on the routing setup, a node might not be able to observe the peer's
reflexive transport address, and attempts to do so might reveal details about
the internal network. In these cases, the node SHOULD NOT offer to provide
address observations.

On-path attackers could capture packets sent from the requester to the
responder, and resend them from a spoofed source address. If done repeatedly,
these spoofed packets could trigger the sending of a large number of OBSERVED_ADDRESS frames.
The recommendation to only include OBSERVED_ADDRESS frames in packets
sent on the same path over which the address was observed ensures

> that the peer will not receive the OBSERVE_ADDRESS frames if the

Reviewer: ```suggestion
that the peer will not receive the OBSERVED_ADDRESS frames if the
```

> addresses are not valid, but does not reduce the number of

Reviewer: ```suggestion
addresses are not valid, but this does not reduce the number of
```

packets sent over the network.
The attack also has the effect of causing spurious
detection NAT rebinding, and is a variant of the replacement of addresses
of packets mentioned in {{Section 21.1.1.3 of RFC9000}}.

> We expect that implementations of QUIC have sufficient

Reviewer: ```suggestion
QUIC implementations are expected to have sufficient
```

protection against spurious NAT rebinding to limit the incidental traffic

> caused by such attacks, and will not produce a high volume of
> spurious OBSERVED_ADDRESS frames.

Reviewer: ```suggestion
caused by such attacks. The same protection logic SHOULD be used to prevent sending of a large number of
spurious OBSERVED_ADDRESS frames.
```
